app/components/progress-updater.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+class ProgressUpdater {
+
+  constructor() {
+
+    this.state = {
+      playing: false,
+      progressObject: null,
+      progressPosition: '0%',
+      subjectViewer: null
+    };
+
+    this.progressUpdateListener = this.handleProgressUpdate.bind(this);
+    this.progressMarkerRenderer = this.renderProgressMarker.bind(this);
+    this.progressMonitoredObject = this.registerProgressingObject.bind(this);
+  }
+
+  setSubjectViewer(subjectViewer){
+    this.state.subjectViewer = subjectViewer;
+  }
+
+  updateSubjectViewer(){
+    if(this.state.subjectViewer){
+      this.state.subjectViewer.forceUpdate();
```

Comment:
Calling `forceUpdate` on a component is bad practice. A component should react to external changes via props, so a better pattern here might be a callback to the subject viewer.